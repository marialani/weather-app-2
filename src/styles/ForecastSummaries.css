.forecast-summaries {
    display: flex;
    justify-content: space-between;
    font-size: 20px;
    padding: 20px;
}