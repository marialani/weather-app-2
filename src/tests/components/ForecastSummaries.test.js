import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 3333444,
        description: "stub description 1",
        icon: 800,
        temperature: {
          max: 22,
          min: 14,
        },
      },
      {
        date: 3344245,
        description: "stub description 2",
        icon: 602,
        temperature: {
          max: 35,
          min: 10,
        },
      },
    ],
    onForecastSelect: () => {},
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
