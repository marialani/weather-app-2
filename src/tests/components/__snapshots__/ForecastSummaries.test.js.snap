// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub description 1
      </div>
      <div
        class="forecast-details__button"
      >
        <button
          class="moreDetails__button"
          type="button"
        >
          More details
        </button>
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        35°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub description 2
      </div>
      <div
        class="forecast-details__button"
      >
        <button
          class="moreDetails__button"
          type="button"
        >
          More details
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
